### PR TITLE
Filter out comments from logged configuration

### LINKED ISSUE #3248: Logging configuration logs comments
When logging the configuration the comments are also logged, meaning that we are logging a lot of information that is not relevant, we should filter out as much as possible before logging.

### PR DESCRIPTION
**Pull request by oyvindeide**
**Issue**
Resolves #3248

**Approach**
Do some checks to filter out comments, it will not get rid of everything, but should get rid of most comments. 

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: tests/ert_tests/shared/test_main_entry.py [hunk 1]
```diff
@@ -120,3 +123,103 @@ def test_vis_database_url_forwarded(monkeypatch):
...
+    base_content = "Content of the configuration file (file_name):\n{}\n"
+    logger = logging.getLogger(__name__)
+    config_path = "file_name"
+    with patch("builtins.open", mock_open(read_data=config_content)), patch(
+        "os.path.isfile", MagicMock(return_value=True)
+    ):
+        with caplog.at_level(logging.INFO):
+            log_config(config_path, logger)
+    expected = base_content.format(expected)
+    result = caplog.text[-len(expected) :]
```
Comment (jondequinor) [inline tests/ert_tests/shared/test_main_entry.py:153]: Maybe use `caplog.messages` instead? Adornments is probably not of consequence here, we care only about the contents.

**Review by jondequinor — COMMENTED** [review]
Why are the tests failing?

File: tests/ert_tests/shared/test_main_entry.py (hunk 1 above)
Comment (oyvindeide) [reply tests/ert_tests/shared/test_main_entry.py]: Forgot about `message`, that makes a lot of sense, updated :+1:

**Comment by oyvindeide** [comment]
> Why are the tests failing?

They are not anymore, was just formatting thing, fixed by using `messages`

> Why are the tests failing?

Because `mock_open` is not [iterable](https://docs.python.org/3/library/unittest.mock.html#mock-open) on py36, so I have elected to skip the test as it only relates to the test. The integration test shows that the production code works as expected.

**Comment by jondequinor** [comment]
Thanks for figuring that out!

**Pull request merged**